examples/collector/src/rando.cc

```diff
@@ -41,172 +47,180 @@ enum supported_types
+        }
+    };
+    return metrics;
 }
 
 void Rando::collect_metrics(std::vector<Metric> &metrics) {
-  std::vector<Metric>::iterator mets_iter;
-  unsigned int seed = time(NULL);
-  int random_value = rand_r(&seed) % 1000;
+    std::vector<Metric>::iterator mets_iter;
+    unsigned int seed = time(NULL);
+    int random_value = rand_r(&seed) % 1000;
 
-  for (mets_iter = metrics.begin(); mets_iter != metrics.end(); mets_iter++) {
-    std::string ns_mts_type = mets_iter->ns()[4].value;
-    int mts_type = metric_types[ns_mts_type];
+    for (mets_iter = metrics.begin(); mets_iter != metrics.end(); mets_iter++) {
+        std::string ns_mts_type = mets_iter->ns()[4].value;
+        int mts_type = metric_types[ns_mts_type];
 
-    switch(mts_type) {
-    case supported_types::float32:
-        mets_iter->set_data((float)random_value);
-        break;
-    case supported_types::float64:
-        mets_iter->set_data((double)random_value);
-        break;
-    case supported_types::int32:
-        mets_iter->set_data((int32_t)random_value);
-        break;
-    case supported_types::int64:
-        mets_iter->set_data((int64_t)random_value);
-        break;
-    case supported_types::uint32:
-        mets_iter->set_data((uint32_t)random_value);
-        break;
-    case supported_types::uint64:
-        mets_iter->set_data((uint64_t)random_value);
-        break;
-    case supported_types::boolean:
-        mets_iter->set_data(random_value%2 ? true : false);
-        break;
-    case supported_types::string:
-        mets_iter->set_data(std::to_string(random_value));
-        break;
+        switch(mts_type) {
+        case supported_types::float32:
+            mets_iter->set_data((float)random_value);
+            break;
+        case supported_types::float64:
+            mets_iter->set_data((double)random_value);
+            break;
+        case supported_types::int32:
+            mets_iter->set_data((int32_t)random_value);
+            break;
+        case supported_types::int64:
+            mets_iter->set_data((int64_t)random_value);
+            break;
+        case supported_types::uint32:
+            mets_iter->set_data((uint32_t)random_value);
+            break;
+        case supported_types::uint64:
+            mets_iter->set_data((uint64_t)random_value);
+            break;
+        case supported_types::boolean:
+            mets_iter->set_data(random_value%2 ? true : false);
+            break;
+        case supported_types::string:
+            mets_iter->set_data(std::to_string(random_value));
+            break;
+        }
+        mets_iter->set_timestamp();
     }
-    mets_iter->set_timestamp();
-  }
 }
 
-int main() {
-  Meta meta(Type::Collector, "rando", 1);
-  Rando plg = Rando();
-  start_collector(&plg, meta);
+int main(int argc, char **argv) {
+    Flags cli;
+    cli.SetFlags();
+    cli.ParseFlags(argc, argv);
+    
+    Meta meta(Type::Collector, "rando", 1);
+
+    if (cli.GetFlagsVM().count("version")) {
```

Comment:
Move this section after the flags parsing - no need to create meta  object if we are just going to exit.